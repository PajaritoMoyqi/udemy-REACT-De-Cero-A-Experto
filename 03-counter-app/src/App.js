// return html to index.js(padre)

import React from 'react'; // to use jsx
import propTypes from 'prop-types';

const App = ({ saludo, subtitulo }) => {
  if(!saludo){
    throw new Error('El saludo es necesario');
  }

  return (
    <>
      <h1> { saludo } </h1>
      <p> { subtitulo } </p>
    </>
  );
}

// need property from index.js

App.propTypes = {
  saludo: propTypes.string.isRequired
}  // variable condition
App.defaultProps = {
  subtitulo: 'Mi Primera Aplicacion'
}  // variable default. no need any import. why?

export default App; // export App.js