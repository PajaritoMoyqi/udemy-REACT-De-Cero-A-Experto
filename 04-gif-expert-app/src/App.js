import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const App = ({defaultCat = []}) => {

  // const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
  const [categories, setCategories] = useState(defaultCat);

  const handleAdd = () => {
    const name = '충사';

    setCategories(cats => [...cats, name]);
  }
  
  return  (
    <>
      <h2>Gif-Expert-App</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        { categories.map((item)=>
          <GifGrid key={item} category={item}
        />) }
      </ol>
    </>
  );

} // key는 무조건 유니크해야 함.

export default App;