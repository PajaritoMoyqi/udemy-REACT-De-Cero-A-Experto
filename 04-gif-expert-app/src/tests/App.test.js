import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import React from "react";
import App from '../App';

describe('App test', () => {


  test('snapshot', () => {
    const wrapper = shallow(<App />)

    expect(wrapper).toMatchSnapshot();
  })
  
  test('show list of categories', () => {
    const categories = ['Hola', 'GG'];
    const wrapper = shallow(<App defaultCat={categories} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  })
  
})
