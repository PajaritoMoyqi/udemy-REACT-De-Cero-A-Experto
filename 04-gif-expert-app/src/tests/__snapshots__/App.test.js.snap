// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test show list of categories 1`] = `
<Fragment>
  <h2>
    Gif-Expert-App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Hola"
      key="Hola"
    />
    <GifGrid
      category="GG"
      key="GG"
    />
  </ol>
</Fragment>
`;

exports[`App test snapshot 1`] = `
<Fragment>
  <h2>
    Gif-Expert-App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
